app/dao/returned_letters_dao.py

```diff
@@ -124,3 +124,37 @@ def fetch_returned_letters(service_id, report_date):
         results = results + query.all()
     results = sorted(results, key=lambda i: i.created_at, reverse=True)
     return results
+
+
+def fetch_returned_letter_callback_data_dao(notification_id, service_id):
+    for table in [Notification, NotificationHistory]:
```

Comment:
With the way the `result` variable is assigned inside the `for` loop, the result from `NotificationHistory` will always overwrite the result from the `Notification` table